mlflow/pyspark/ml/__init__.py

```diff
@@ -1120,20 +1111,63 @@ def _infer_model_signature(input_example_slice):
                     ).toPandas()
                 else:
                     input_example = None
-                mlflow.spark.log_model(
+                spark_model_id = mlflow.spark.log_model(
                     spark_model,
                     "model",
                     registered_model_name=registered_model_name,
                     input_example=input_example,
                     signature=signature,
-                )
+                ).model_id
                 if _is_parameter_search_model(spark_model):
-                    mlflow.spark.log_model(
+                    best_model_id = mlflow.spark.log_model(
                         spark_model.bestModel,
                         "best_model",
-                    )
+                    ).model_id
             else:
                 _logger.warning(_get_warning_msg_for_skip_log_model(spark_model))
+        if _is_parameter_search_estimator(estimator):
```

Comment:
This part is mostly switching the order with log_model so that we can have the model_id ready when we log metrics